kornia-py/src/lib.rs

```diff
@@ -24,66 +24,289 @@ pub fn get_version() -> String {
-    m.add_function(wrap_pyfunction!(icp::icp_vanilla, m)?)?;
-    m.add_function(wrap_pyfunction!(io::functional::read_image_any, m)?)?;
-    m.add_function(wrap_pyfunction!(io::functional::read_image, m)?)?;
-    m.add_function(wrap_pyfunction!(io::png::decode_image_png_u8, m)?)?;
-    m.add_function(wrap_pyfunction!(io::png::decode_image_png_u16, m)?)?;
-    m.add_function(wrap_pyfunction!(io::png::read_image_png_u8, m)?)?;
-    m.add_function(wrap_pyfunction!(io::png::read_image_png_u16, m)?)?;
-    m.add_function(wrap_pyfunction!(io::png::write_image_png_u8, m)?)?;
-    m.add_function(wrap_pyfunction!(io::png::write_image_png_u16, m)?)?;
-    m.add_function(wrap_pyfunction!(io::jpeg::decode_image_jpeg, m)?)?;
-    m.add_function(wrap_pyfunction!(io::jpeg::read_image_jpeg, m)?)?;
-    m.add_function(wrap_pyfunction!(io::jpeg::write_image_jpeg, m)?)?;
-    m.add_function(wrap_pyfunction!(io::jpeg::encode_image_jpeg, m)?)?;
-    m.add_function(wrap_pyfunction!(io::tiff::read_image_tiff_f32, m)?)?;
-    m.add_function(wrap_pyfunction!(io::tiff::read_image_tiff_u8, m)?)?;
-    m.add_function(wrap_pyfunction!(io::tiff::read_image_tiff_u16, m)?)?;
-    m.add_function(wrap_pyfunction!(io::tiff::write_image_tiff_f32, m)?)?;
-    m.add_function(wrap_pyfunction!(io::tiff::write_image_tiff_u8, m)?)?;
-    m.add_function(wrap_pyfunction!(io::tiff::write_image_tiff_u16, m)?)?;
-    m.add_function(wrap_pyfunction!(io::jpegturbo::decode_image_jpegturbo, m)?)?;
-    m.add_function(wrap_pyfunction!(io::jpegturbo::read_image_jpegturbo, m)?)?;
-    m.add_function(wrap_pyfunction!(io::jpegturbo::write_image_jpegturbo, m)?)?;
-    m.add_function(wrap_pyfunction!(resize::resize, m)?)?;
-    m.add_function(wrap_pyfunction!(warp::warp_affine, m)?)?;
-    m.add_function(wrap_pyfunction!(warp::warp_perspective, m)?)?;
-    m.add_class::<PyImageDecoder>()?;
-    m.add_class::<PyImageEncoder>()?;
+
+    // Deprecated root-level functions
+    m.add_function(wrap_pyfunction!(rgb_from_gray, m)?)?;
+    m.add_function(wrap_pyfunction!(rgb_from_rgba, m)?)?;
+    m.add_function(wrap_pyfunction!(rgb_from_bgra, m)?)?;
+    m.add_function(wrap_pyfunction!(bgr_from_rgb, m)?)?;
+    m.add_function(wrap_pyfunction!(gray_from_rgb, m)?)?;
+    m.add_function(wrap_pyfunction!(add_weighted, m)?)?;
+    m.add_function(wrap_pyfunction!(compute_histogram, m)?)?;
+    m.add_function(wrap_pyfunction!(icp_vanilla, m)?)?;
+    m.add_function(wrap_pyfunction!(read_image_any, m)?)?;
+    m.add_function(wrap_pyfunction!(read_image, m)?)?;
+    m.add_function(wrap_pyfunction!(read_image_jpeg, m)?)?;
+    m.add_function(wrap_pyfunction!(write_image_jpeg, m)?)?;
+    m.add_function(wrap_pyfunction!(read_image_png_u8, m)?)?;
+    m.add_function(wrap_pyfunction!(write_image_png_u8, m)?)?;
+    m.add_function(wrap_pyfunction!(resize, m)?)?;
+    m.add_function(wrap_pyfunction!(warp_affine, m)?)?;
+    m.add_function(wrap_pyfunction!(warp_perspective, m)?)?;
```

Comment:
there's are some missing functions not raising deprecation warninig which will break backwards compatibility